File: code.eval/02-moderate/pangrams/pangrams.py
# Reuben Thorpe (2016), CodeEval [Pangrams v1.0]
from sys import argv
from string import ascii_lowercase


def parse_problem(file_path):
    with open(argv[1], 'r') as in_file:
        for line in in_file:
            yield(set(char for char in line.strip().lower().replace(' ','')))





if __name__ == '__main__':
    alphabet = set(ascii_lowercase)

    for problem in parse_problem(argv[1]):
        solution = ''.join(sorted(char for char in
                                  alphabet.difference(problem)))

        print(solution) if solution else print('NULL')

File: code.eval/02-moderate/pangrams/pangrams.cpp
// Reuben Thorpe (2016), CodeEval [Pangrams v1.1]
#include <iostream>
#include <fstream>
#include <vector>
#include <algorithm>

using namespace std;

string missing_lowercase_ascii_alpha(string line, string &alpha_lower);


int main(int argc, char* argv[]) {
  ifstream in_file(argv[1]);
  string line;
  string alpha_lower = "abcdefghijklmnopqrstuvwxyz";

  while (getline(in_file, line)) {
  cout << missing_lowercase_ascii_alpha(line, alpha_lower) << '\n';
  }

  return(0);
}


string missing_lowercase_ascii_alpha(string line, string &alpha_lower) {
  // Determin what letters from the lowercase ascii are missing from the string.

  transform(line.begin(), line.end(), line.begin(), ::tolower);
  sort(line.begin(), line.end());
  auto last = unique(line.begin(), line.end());
  line.erase(last, line.end());
  line.erase(line.begin(), line.begin()+1);

  string solution;

  set_difference(alpha_lower.begin(), alpha_lower.end(), line.begin(),
                 line.end(), back_inserter(solution));

  if (solution.empty()) return("NULL");
  else return(solution);

}


